From sarah.a.sharp@linux.intel.com  Fri Oct  9 10:39:12 2009
From: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Date: Fri, 18 Sep 2009 08:55:12 -0700
Subject: USB: xhci: Fix dropping endpoints from the xHC schedule.
To: Greg KH <gregkh@suse.de>
Cc: linux-usb@vger.kernel.org
Message-ID: <20090918155512.GA3876@xanatos>
Content-Disposition: inline


When an endpoint is to be dropped from the hardware bandwidth schedule, we
want to clear its add flag.

Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/xhci-hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/xhci-hcd.c
+++ b/drivers/usb/host/xhci-hcd.c
@@ -895,7 +895,7 @@ int xhci_drop_endpoint(struct usb_hcd *h
 	ctrl_ctx->drop_flags |= drop_flag;
 	new_drop_flags = ctrl_ctx->drop_flags;
 
-	ctrl_ctx->add_flags = ~drop_flag;
+	ctrl_ctx->add_flags &= ~drop_flag;
 	new_add_flags = ctrl_ctx->add_flags;
 
 	last_ctx = xhci_last_valid_endpoint(ctrl_ctx->add_flags);
